common/state/state_kv.h

```diff
@@ -25,24 +25,24 @@ namespace state
     {
         typedef enum
         {
+            KV_UNINITIALIZED,
             KV_CREATE,
             KV_OPEN
         } kv_start_mode_e;
 
     protected:
         pdo::state::StateNode rootNode_;
         const ByteArray state_encryption_key_;
-        data_node_io dn_io_;
-        kv_start_mode_e kv_start_mode;
+        mutable data_node_io dn_io_;
```

Comment:
i don't think i've ever seen a place where mutable was useful. this might be a first.

Reply:
Same here -- always learning!